src/components/base/QRCode/index.js

```diff
@@ -32,7 +33,8 @@ class QRCode extends PureComponent<Props> {
   }
 
   render() {
-    return <canvas ref={n => (this._canvas = n)} />
+    const { hidden } = this.props
+    return <canvas hidden={hidden} ref={n => (this._canvas = n)} />
```

Comment:
would be more DOM friendly to only update opacity style (that would require to "stack" all the qr codes in absolute)

Reply:
`hidden` is part of DOM right? not sure which is more efficient, opacity or hidden. position:absolute might help not having the browser do a position calc but I'm pretty sure it's premature optim. it would complexity the parent code to do absolute position

Reply:
Yeah it's part of the DOM, that's not what I meant. But `hidden` trigger layout update while `opacity` trigger only a paint one. I would not underestimate that difference nor call it premature opti, mostly in that case where we will draw qrcode multiple time / s. Having a relative wrapper with fixed height and positionning qrcodes in absolute is not so complex compared to the perf gain.

relatable:
https://codepen.io/meriadec/pen/NYNrpY 
left square: hidden
right square: opacity

Reply:
Following our discussion, I will do the update myself after the merge. Can you rebase your branch from master?